RNAdam-core/src/main/scala/org/bdgenomics/RNAdam/algorithms/quantification/Tare.scala

```diff
@@ -0,0 +1,135 @@
+/**
+ * Licensed to Big Data Genomics (BDG) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The BDG licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.RNAdam.algorithms.quantification
+
+import org.apache.spark.SparkContext._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.mllib.linalg.Vectors
+import org.apache.spark.mllib.regression.{ LabeledPoint, LinearRegressionWithSGD }
+import scala.math.{ exp, log }
+
+object Tare extends Serializable {
+
+  /**
+   * Converts a base into an integer number. Only works for A/C/G/T, both
+   * lower and upper case. Will throw an exception for other bases, user is
+   * expected to check before passing to function.
+   *
+   * @param base The base to convert into an integer.
+   * @return Returns a mapping from a base to an integer.
+   */
+  private def idx(base: Char): Int = base match {
```

Comment:
I'm not saying there's a change to be made here now, but this is yet another use-case for having a separate 'alphabet' style value, with encodings and conversions for individual letters.